fiftyone/brain/__init__.py

```diff
@@ -703,3 +703,99 @@ def compute_exact_duplicates(
     return fbd.compute_exact_duplicates(
         samples, num_workers, skip_failures, progress
     )
+
+
+def compute_leaky_splits(
+    samples,
+    brain_key=None,
+    split_views=None,
+    split_field=None,
+    split_tags=None,
+    threshold=0.2,
+    similarity_brain_key=None,
+    embeddings_field=None,
```

Comment:
Currently, this code results in all sample embeddings being computed:

```py
view1 = dataset.limit(500)
view2 = dataset.skip(500).limit(500)

fob.compute_leaky_splits(dataset, split_views={"train":view1, "test":view2}, brain_key="leak_key2", embeddings_field="clip_embeddings")
```

When in practice only 1000 need embeddings. I think this is fine, just calling it out

Reply:
See comment below.